lib/daru/category.rb

```diff
@@ -918,10 +918,28 @@ def modify_category_at pos, category
       end
       old_category = category_from_position pos
       @array[pos] = int_from_cat category
-      @cat_hash[old_category].delete pos
+      @cat_hash[old_category].delete pos unless old_category.nil?
       @cat_hash[category] << pos
     end
 
+    def set pos, category
+      unless categories.include? category
+        raise ArgumentError, "Invalid category #{category}, "\
+          'to add a new category use #add_category'
+      end
+      pos = pos.is_a?(Array) ? pos : [pos]
+      pos.each do |x|
+        if @index.include? x
+          self.[]= x, category
+          next
+        end
+        @array << categories.index(category)
+        @index = @index.add(x)
+        @cat_hash.store(category,[x])
+        self.[]= x, category
```

Comment:
Why are you calling `self.[]= x, category` at line 939?

Reply:
This line is added to make sure all other variables are updated which would be the case if you set the index usually.

Reply:
That seems like an over kill. There are just two variables which needs updating `@array` and `@cat_hash`. You have updated the `@array` and for `@cat_hash`, you can just do `@cat_hash[category] << pos`.